src/models/activity.model.js

```diff
@@ -14,5 +14,24 @@ module.exports = (() => {
     ]
   });
 
+  ActivitySchema.pre('deleteMany', function() {
+    const { _conditions } = this;
+    return deleteSubdocumenets(_conditions);
+  });
+
+  function deleteSubdocumenets(conditions) {
+    const ActivityModel = mongoose.model('ActivityModel');
+    const SubreminderModel = mongoose.model('SubreminderModel');
+
+    return ActivityModel.find(conditions)
+      .then(activities =>
+        activities.reduce((prev, curr) => prev.concat(curr.subreminders), [])
+      )
+      .then(subreminders => subreminders.map(s => s._id))
+      .then(subreminderIds =>
+        SubreminderModel.deleteMany({ _id: { $in: subreminderIds } })
+      );
+  }
```

Comment:
instead of using this way to delete sub documents, checkout https://stackoverflow.com/questions/11904159/automatically-remove-referencing-objects-on-deletion-in-mongodb
maybe this can reduce lines of code

Reply:
Won't work in this case. You need to have 1. reminder referenced in event/activity schema.  2. reminder referenced in subreminder schema

Reply:
Also, the method you mentioned(i.e., Model.remove) is deprecated in mongoose. So there's no point to use that.

Reply:
You need deleteMany either way.